src/generic/ResidueField.jl

```diff
@@ -668,3 +668,14 @@ end
 function  gen(R::AbstractAlgebra.Generic.ResField{AbstractAlgebra.Generic.Poly{Rational{BigInt}}})
    return R(gen(base_ring(R)))
 end
+
+function RandomExtensions.make(S::AbstractAlgebra.ResField{AbstractAlgebra.Generic.Poly{Rational{BigInt}}}, vs...)
```

Comment:
Does it have to be such a constrained signature? Wouldn't it be confusing to get a different behavior if you have e.g. `Int` instead of `BigInt` ? Or a `PolyElem` instead of a `Generic.Poly` ?

Reply:
No, I guess it doesn't have to be so constrained. The fix was mainly for NumberField elements, which for now doesn't have a separate type. That would eventually be fixed, and these rand functions would be converted over to functions for a future AbstractAlgebra.NumberField type.

It's really an independent question whether we want to omit the extra argument for residue rings over polynomial rings. This can probably be done for (univariate) polynomial rings over a field. I'm not sure whether it works more generally than that.

I can change it to PolyElem{T} where T <: Field I guess.

Reply:
The main issue is that there are then other types of rings over polynomial rings over fields where the extra argument to rand still exists. So we'd have to change them all. This seems like a different ticket to me and unrelated to this one.

Granted, the best way to fix this issue would be to add a proper number field type (it would also fix the fact that the string for the generator of the number field is currently ignored by the NumberField constructor because there's nowhere to put it inside the ResidueField type). But that is also a different ticket, and not a priority.

I'm in two minds about loosening the restriction here.

Reply:
I think it makes sense to allow `T <: FieldElem`.

Reply:
I guess you meant T <: FieldElement. I've made that change.

Reply:
Actually, this doesn't work. This passes an additional argument to residue fields over polynomial rings over a Galois field.

I think it is better to leave the signature restricted.

Reply:
Would you mind giving the example which doesn't work to help me understand? For me it's not blocking the PR, but I would like to understand better to which extent we can add methods like this for other residue fields, like
```
R = Generic.ResidueField(ZZ, 7)
# or
R, x = PolynomialRing(Nemo.QQ, "x")
K = ResidueField(R, x^3 + 3x + 1)
```

Reply:
Our current test code fails with that change. Unfortunately Windows updated my computer (unsuccessfully) in the mean time, so the backtrace is now lost. But you can just make the change you wanted and you will see the test code fails.